runtime/configs/config.go

```diff
@@ -15,11 +27,62 @@ type Config struct {
 	Telemetry   *TelemetryConfig   `json:"telemetry"`
 	Logger      *LoggerConfig      `json:"logger"`
 	RPC         *RPCConfig         `json:"rpc"`
+	Keybase     *KeybaseConfig     `json:"keybase"`
+}
+
+// ParseConfig parses the config file and returns a Config struct
+func ParseConfig(cfgFile string) *Config {
+	config := NewDefaultConfig()
+
+	if cfgFile != "" {
+		viper.SetConfigFile(cfgFile)
+	} else {
+
+		viper.AddConfigPath("/etc/pocket/")  // path to look for the config file in
+		viper.AddConfigPath("$HOME/.pocket") // call multiple times to add many search paths
+		viper.AddConfigPath(".")             // optionally look for config in the working directory
+		viper.SetConfigName("config")        // name of config file (without extension)
+		viper.SetConfigType("json")          // REQUIRED if the config file does not have the extension in the name
+	}
+
+	// The lines below allow for environment variables configuration (12 factor app)
+	// Eg: POCKET_CONSENSUS_PRIVATE_KEY=somekey would override `consensus.private_key` in config.json
+	// If the key is not set in the config, the env var will not be used.
+	viper.SetEnvPrefix("POCKET")
+	viper.SetEnvKeyReplacer(strings.NewReplacer(".", "_"))
+	viper.AutomaticEnv()
+
+	if err := viper.ReadInConfig(); err != nil {
+		if _, ok := err.(viper.ConfigFileNotFoundError); ok && cfgFile == "" {
+			// if file does not exist and there is no override, return default config
+			// DISCUSS: how should we handle this? Should we return an error? Should we create the default file?
+			return config
+		} else {
+			// Some other error occurred while reading the config file
+			log.Fatalf("[ERROR] failed to read config %s", err.Error())
```

Comment:
I think this was added before we had a logger.

Can you change these to `logger.Global.Error(err).Fatalf("failed to read config")`. Ditto below

Reply:
Need to move things around with our logger_config.proto first. This causes import cycles.

Reply:
Just leave a TODO so it's not a blocker for this PR.